Sources/VLCOpenNetworkStreamViewController.m

```diff
@@ -178,70 +178,66 @@ - (IBAction)goBack:(id)sender
-                                                              object:textField
-                                                               queue:[NSOperationQueue mainQueue]
-                                                          usingBlock:^(NSNotification * _Nonnull note) {
-                okAction.enabled = (textField.text.length != 0);
-            }];
-        }];
-
-        [alertController addAction:cancelAction];
-        [alertController addAction:okAction];
+        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"BUTTON_OK", nil)
+                                                           style:UIAlertActionStyleCancel
+                                                         handler:nil];
 
+        [alertController addAction:okAction];
         [self presentViewController:alertController animated:YES completion:nil];
+        return;
     }
-    else
-    {
-        UIAlertView *alertView = [[UIAlertView alloc] init];
-        alertView.delegate = self;
-        alertView.alertViewStyle = UIAlertViewStylePlainTextInput;
-        alertView.title = renameString;
-        alertView.tag = cellIndexPath.row; // Dirty...
-        [alertView addButtonWithTitle:cancelString];
-        [alertView addButtonWithTitle:@"OK"];
-        [alertView show];
+    if (!self.privateToggleSwitch.on) {
+        NSString *urlString = self.urlField.text;
+        if ([_recentURLs indexOfObject:urlString] != NSNotFound)
+            [_recentURLs removeObject:urlString];
+
+        if (_recentURLs.count >= 100)
+            [_recentURLs removeLastObject];
+        [_recentURLs addObject:urlString];
+        [[NSUbiquitousKeyValueStore defaultStore] setArray:_recentURLs forKey:kVLCRecentURLs];
+
+        [self.historyTableView reloadData];
     }
+    [self.urlField resignFirstResponder];
+    [self _openURLStringAndDismiss:self.urlField.text];
+}
+
+- (void)renameStreamFromCell:(UITableViewCell *)cell
+{
+    NSIndexPath *cellIndexPath = [self.historyTableView indexPathForCell:cell];
+
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"BUTTON_RENAME", nil)
+                                                                             message:nil
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+
+    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"BUTTON_CANCEL", nil)
+                                                           style:UIAlertActionStyleCancel
+                                                         handler:nil];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        NSString *streamTitle = alertController.textFields.firstObject.text;
+        [self renameStreamWithTitle:streamTitle atIndex:cellIndexPath.row];
```

Comment:
I don't think we need `streamTitle`, we could directly use `alertController.textFields.firstObject.text` but it could be debatable readability-wise 😄

Reply:
yes I think it's actually nicer with the streamTitle readability wise and it provides the context that it is the new streamtitle